Source/WebKit/Shared/mac/WebEventFactory.mm

```diff
@@ -347,7 +347,11 @@ static int typeForEvent(NSEvent *event)
     double pressure = [event type] == NSEventTypePressure ? event.pressure : lastPressureEvent.pressure;
     double force = pressure + stage;
 
-    return WebMouseEvent({ type, modifiers, timestamp, WTF::UUID::createVersion4() }, button, buttons, WebCore::IntPoint(position), WebCore::IntPoint(globalPosition), deltaX, deltaY, deltaZ, clickCount, force, WebMouseEventSyntheticClickType::NoTap, eventNumber, menuTypeForEvent);
+    CGEventRef cgEvent = [event CGEvent];
```

Comment:
Is the NSEvent guaranteed to have a CGEvent? I suspect some synthesized mouse events may not.

Reply:
Also why doesn't this call `unadjustedMovementForEvent()`?

Reply:
> Is the NSEvent guaranteed to have a CGEvent? I suspect some synthesized mouse events may not.

The event parameter to CGEventGetIntegerValueField is nullable and it will return 0 if event is NULL.

> Also why doesn't this call unadjustedMovementForEvent()?

I don't know the history of the duplication between WebEventFactory and PlatformEventFactoryMac, but there is quite a lot of it, and I didn't feel up to the task of unification. I see [this comment](https://github.com/WebKit/WebKit/blob/c68f7caf5b378de2ea3e9b95e9745ada7df61203/Source/WebCore/platform/mac/PlatformEventFactoryMac.h#L48) in PlatformEventFactoryMac:

```
// FIXME: WebKit2 has a lot of code copied and pasted from PlatformEventFactoryMac in WebEventFactory. More of it should be shared with WebCore.
```

Is your suggestion that we add this to PlatformEventFactoryMac.h and then call that from WebEventFactory.mm?

```
WEBCORE_EXPORT IntPoint unadjustedMovementForEvent(NSEvent *event);
```

Reply:
Done in my latest push.